Basic Compact Interactive List documentation (#1984)
cerner/terra-framework: packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/BasicCompactInteractiveList.1.doc.mdx

import { Notice } from '@cerner/terra-docs';
import BasicCompactInteractiveList from './BasicCompactInteractiveList?dev-site-examples';
import ResponsiveColumns from './ResponsiveColumns?dev-site-examples';

# Basic Compact Interactive List

### Description

* The **[Compact Interactive List](/components/cerner-terra-framework-docs/compact-interactive-list/about)** component consists of unique rows. 
  Each row has a border, which visually separates it from the other rows in the list.
* The rows are visually stractured in semantic columns, which do not have visual separators, but provide data alighnment and allow [**keyboard navigation**](/components/cerner-terra-framework-docs/compact-interactive-list/about#keyboard-interactions). 
  The **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) property** provides the rules for semantic columns structure. 

> * The **[cells](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** provides the unique row content. 

Reviewer: Does it necessarily have to be unique? By design, most columns will contain similar content in each row.
```suggestion
* The **[cells](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** provides the row content. 
```

>   The number of cells must be exactly the same, as the number of elements in **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) array**. 

Reviewer: ```suggestion
  The number of cells in a row must be exactly the same, as the number of elements in **[columns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) array**. 
```

  If a cell is supposed to be empty, the **[content](/components/cerner-terra-framework-docs/compact-interactive-list/about#row) property** must be an empty string.

### Creating Basic Compact Interactive List

#### Basic Columns Property Example

> * Break row data into semantic columns in a meaningful way. Each semantic column should have as little elements as possible. 

Reviewer: ```suggestion
* Break row data into semantic columns in a meaningful way. Each semantic column should have as few elements as possible. 
```

  In the example below, the data is split in following columns: `Icon`, `Service Name`, and `Featured`.
* Provide unique **[id](/components/cerner-terra-framework-docs/compact-interactive-list/about#column)**
  and meaningful **[displayName](/components/cerner-terra-framework-docs/compact-interactive-list/about#column)**

>   to all semantic columns. The column headers are hidden, but they will be anounced with the assistive technologies to provide the column context.

Reviewer: ```suggestion
  to all semantic columns. The column headers are hidden, but they will be used by assistive technologies, like screen readers, to provide column context.
```

> * Provide meaningful row headers, as assistive technologies announce the row header cell content per row context change. 

Reviewer: ```suggestion
* Provide meaningful, identifying row headers. Assistive technologies use these to provide context as a user navigates through rows.
```

``` js
const cols = [
  {
    id: 'Column-0',
    displayName: 'Icon',
    width: '60px', // width property is not required, it is used to provide a better visual structure
  },
  {
    id: 'Column-1',
    displayName: 'Service name', // this column info will be announced as a row header
    // omitting width property will allow the semanic column to flex grow/shrink
  },
  {
    id: 'Column-2',
    displayName: 'Featured',
    width: '60px', // width property is not required, it is used to provide a better visual structure
  },
];
```

#### Basic Rows Prop Example

This example demonstrates basic usage of the **[rows](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) property**. 

``` js
const rows = [
  {
    id: 'row_1',
    cells: [
      { content: iconResultsNormal },
      { content: 'Discern Care Set (1)' },
      { content: <FeaturedIcon /> },
    ],
  },
  ... // 4 more row elements
];
```

#### Basic Compact Interactive List Example

> This example demonstrates a basic Compact Interactive List component that consists of 5 semantic rows, organised in 3 semantic columns, where the `Service Name` column is a row header.

Reviewer: ```suggestion
This example demonstrates a basic Compact Interactive List component that consists of 5 semantic rows, organized in 3 semantic columns, where the `Service Name` column is a row header.
```

``` js
const DefaultCompactInteractiveList = () => (
  <CompactInteractiveList

>     id="default-conpact-interactive-list"

Reviewer 2: ```suggestion
    id="default-compact-interactive-list"
```

Author: Good catch!

    ariaLabel="Default Compact Interactive List"
    rows={rows}
    columns={cols}

>     rowHeaderIndex={1} // points out to Service Name column

Reviewer: ```suggestion
    rowHeaderIndex={1} // Specifies 'Service Name' column as row header for every semantic row.
```

  />
);
```

<BasicCompactInteractiveList />

#### Compact Interactive List with Several Visual Columns

> To save space, the rows can be organized in visual columns with **[numberOfColumns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) 

Reviewer: ```suggestion
To save space and facilitate faster keyboard navigation, the rows can be organized in visual columns with **[numberOfColumns](/components/cerner-terra-framework-docs/compact-interactive-list/about#compact-interactive-list-properties-table) 
```
